arm64: dts: qcom: msm8916-vivo-y21l: Add initial device tree

raihan2000 opened the pull request:
Vivo Y21L is a handset using the MSM8916 SoC released in 2016.

Add device tree with initial support for:
 - Accelerometer
 - Ambient Light
 - GPIO keys
 - Magnetormeter
 - SDHCI (internal and external storage)
 - UART
 - USB Device Mode
 - WCNSS (WiFi/BT)
 - Regulators

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-vivo-y21l.dts:
```diff
@@ -0,0 +1,158 @@
+		pinctrl-names = "default";
+	};
+
+	light-sensor@39 {
```
Please order these nodes by reg address (18 -> 2e -> 39)

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-vivo-y21l.dts:
```diff
@@ -0,0 +1,158 @@
+
+&sdhc_1 {
+	vmmc-supply = <&pm8916_l8>;
+	vqmmc-supply = <&pm8916_l5>;
```
You can drop these two -supplys, they are already in `msm8916-pm8916.dtsi`.

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-vivo-y21l.dts:
```diff
@@ -0,0 +1,158 @@
+
+&usb_hs_phy {
+	v1p8-supply = <&pm8916_l7>;
+	v3p3-supply = <&pm8916_l13>;
```
Same for these

stephan-gh reviewed (commented):
Can you change `git config user.name "Raihan Ahamed"`, then `git commit --amend --reset-author`. And put `Raihan Ahamed` instead of `raihan2000` into the Signed-off-by. Full names are preferred for the Signed-off-by.

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-vivo-y21l.dts:
```diff
@@ -0,0 +1,158 @@
+		vddio-supply = <&pm8916_l6>;
+		mount-matrix = "1", "0", "0",
+				"0", "1", "0",
+				"0", "0", "1";
```
I think you can omit this property if you just need an identity matrix. But, please check if the screen rotations are reported correctly. See the instructions using `monitor-sensor` in  https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Testing_2

raihan2000 committed:
arm64: dts: qcom: msm8916-vivo-y21l: Add initial device tree

Vivo Y21L is a handset using the MSM8916 SoC released in 2016.

Add device tree with initial support for:
 - Accelerometer
 - Ambient Light
 - GPIO keys
 - Magnetormeter
 - SDHCI (internal and external storage)
 - UART
 - USB Device Mode
 - WCNSS (WiFi/BT)
 - Regulators

Signed-off-by: Raihan Ahamed <raihan1999ahamed@gmail.com>

stephan-gh commented:
I fixed the indentation of mount-matrix and pushed to `msm8916/6.6`. Thanks!
